lib/edit-site-page.php

```diff
@@ -146,50 +145,31 @@ function gutenberg_edit_site_init( $hook ) {
 		$settings['fontSizes'] = $font_sizes;
 	}
 
-	// Get all templates by triggering `./template-loader.php`'s logic.
-	$template_getters  = array(
-		'get_embed_template',
-		'get_404_template',
-		'get_search_template',
-		'get_home_template',
-		'get_privacy_policy_template',
-		'get_post_type_archive_template',
-		'get_taxonomy_template',
-		'get_attachment_template',
-		'get_single_template',
-		'get_page_template',
-		'get_singular_template',
-		'get_category_template',
-		'get_tag_template',
-		'get_author_template',
-		'get_date_template',
-		'get_archive_template',
-	);
 	$template_ids      = array();
 	$template_part_ids = array();
-	foreach ( $template_getters as $template_getter ) {
-		call_user_func( $template_getter ); // This sets $_wp_current_template_hierarchy.
+	foreach ( get_template_types() as $template_type ) {
+		// Skip 'embed' for now because it is not a regular template type.
+		// Skip 'index' because it's a fallback that we handle differently.
+		if ( in_array( $template_type, array( 'embed', 'index' ), true ) ) {
+			continue;
+		}
 
-		$current_template_post = gutenberg_find_template_post( $_wp_current_template_hierarchy );
+		$template_hierarchy    = array_merge( get_template_hierachy( $template_type ), get_template_hierachy( 'index' ) );
+		$current_template_post = gutenberg_find_template_post( $template_hierarchy );
 		if ( isset( $current_template_post ) ) {
 			$template_ids[ $current_template_post->post_name ] = $current_template_post->ID;
 		}
 		if ( isset( $_wp_current_template_part_ids ) ) {
 			$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
 		}
 
-		$_wp_current_template_hierarchy = null;
-		$_wp_current_template_part_ids  = null;
-	}
-	get_front_page_template();
-	get_index_template();
-	$current_template_post                             = gutenberg_find_template_post( $_wp_current_template_hierarchy );
-	$template_ids[ $current_template_post->post_name ] = $current_template_post->ID;
-	if ( isset( $_wp_current_template_part_ids ) ) {
-		$template_part_ids = $template_part_ids + $_wp_current_template_part_ids;
+		$_wp_current_template_part_ids = null;
 	}
-	$settings['templateId']      = $current_template_post->ID;
-	$settings['homeTemplateId']  = $current_template_post->ID;
+
+	$current_template_id = $template_ids['front-page'];
```

Comment:
Right now, if the block-based theme doesn't have a template file called `front-page.html` it throws a 404 when trying to load the page. Could it be related to this somehow?

Reply:
Found my way back here as well 😄  It's pretty evident in the default `twentytwenty` theme, which doesn't have a `front-page` template.

https://github.com/WordPress/twentytwenty

This causes the Site Editor to not load at all.

Seems it should be resilient to the absence of this template, possibly falling back to another applicable template?

e.g. `front-page`, `home`, `index`

https://developer.wordpress.org/files/2014/10/Screenshot-2019-01-23-00.20.04.png

Reply:
That's possible I'm afraid :grimacing: 
I'm still touching this file pretty regularly, will look into fixing it :+1:

Reply:
> Seems it should be resilient to the absence of this template, possibly falling back to another applicable template?
> 
> e.g. `front-page`, `home`, `index`
> 
> https://developer.wordpress.org/files/2014/10/Screenshot-2019-01-23-00.20.04.png
> https://developer.wordpress.org/themes/basics/template-hierarchy/

Yeah, I'm painfully aware of the template hierarchy :grimacing: The function that's used to populate `$template_ids` actually falls back to `index`, see https://github.com/WordPress/gutenberg/blob/19fb4aed861101bb496697b4886d2b386ad1d9c7/lib/template-loader.php#L226

I'll have to dig into why the `index` fallback isn't working.

Reply:
Fix: #22954